packages/contracts/test/SMT.ts

```diff
@@ -0,0 +1,97 @@
+import { ethers } from 'hardhat'
+import { expect } from 'chai'
+import poseidon from '../src/poseidon'
+import { SparseMerkleTree, hashLeftRight } from '@unirep/crypto'
+
+const getSMT = async (depth = 32) => {
+    const accounts = await ethers.getSigners()
+    const libraries = {}
+    for (const [inputCount, { abi, bytecode }] of Object.entries(
+        poseidon
+    ) as any) {
+        const f = new ethers.ContractFactory(abi, bytecode, accounts[0])
+        const c = await f.deploy()
+        await c.deployed()
+        libraries[`Poseidon${inputCount}`] = c.address
+    }
+    delete libraries['Poseidon3']
+    delete libraries['Poseidon5']
+    const merkleTreeLibFactory = await ethers.getContractFactory(
```

Comment:
Could we move deploy SMT lib to beforeEach?

Reply:
Maybe, this is the same problem as before. There's no way to pass variables from `beforeEach` to the test function. I'd rather use this function to deploy than have shared state variables being mutated.